src/components/SubmissionPage/SubmissionPage.jsx

```diff
@@ -0,0 +1,50 @@
+import { Flex, Input, Button } from "@chakra-ui/react";
+import { useRef, useState, useEffect } from "react";
+
+export const SubmissionPage = () => {
+  const [disableButton, setDisabledButton] = useState(true);
+  const inputRef = useRef();
+
+  function isUrlValid(string) {
+    var res = string.match(
+      /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g
+    );
+    return res !== null;
+  }
+
+  const onChangeHandler = (e) => {
+    if (isUrlValid(inputRef.current.value)) {
+      setDisabledButton(false);
+      console.log("correct");
+    } else {
+      setDisabledButton(true);
+      console.log("please enter the correct URL");
+    }
+  };
+
+  const onClickHandler = () => {
```

Comment:
- same goes here, you can name this function something like `submitPortfolioUrl`

```suggestion
  const submitPortfolioUrl = () => {
```

- One small suggestion: if you are using names like `submitPortfolioUrl` or `checkPortfolioUrl` is itself self explanatory, so we don't have to write "handler" in function names & also, we should avoid them.

Reply:
Yes makes sense, Thanks for the suggestion will remember it for the future as well, Thank you